homeassistant/components/input_boolean/__init__.py

```diff
@@ -153,3 +213,9 @@ def is_on(self):
         """Turn the entity off."""
         self._state = False
         await self.async_update_ha_state()
+
+    async def async_update_config(self, config: typing.Dict) -> None:
+        """Handle when the config is updated."""
+        self._config = config
+        self._state = config.get(CONF_INITIAL, self._state)
+        self.async_schedule_update_ha_state()
```

Comment:
Little faster:
```suggestion
        self.async_write_ha_state()
```

Reply:
I'd love to read more about when/why you would use `async_schedule_update_ha_state()` (and what is still confusing to me, since it is named `async_*` do I await it or not?), `async_write_ha_state()` etc